build-tools/packages/build-tools/src/fluidBuild/fluidPackageCheck.ts

```diff
@@ -488,22 +514,73 @@ export class FluidPackageCheck {
     private static checkLintScripts(pkg: Package, fix: boolean) {
         let fixed = false;
         if (pkg.getScript("build")) {
-            const hasPrettier = pkg.getScript("prettier");
-            const lintChildren = hasPrettier ? ["prettier", "eslint"] : ["eslint"];
-            if (this.checkChildrenScripts(pkg, "lint", lintChildren, false, fix)) {
-                fixed = true;
+            // TODO: add prettier check comment once prettier is enforced globally, hasPrettier commented out to discard build warnings from "lint" & "lint:fix" scripts
+            // const hasPrettier = pkg.getScript("prettier");
+
+            const lintScript = pkg.getScript("lint");
+            const lintFixScript = pkg.getScript("lint:fix");
+
+            /* Regular Expression
+                lintRegex is a regex to extract lint-package name from lintScript & lintFixScript
+
+                if lintScript:
+                    -> lintMatch = ["npm run foo", "npm run bar", "npm run foo-bar"]
+                    -> lintChildren = ["foo", "bar", "foo-bar"]
+                else:
+                    -> lintFixMatch = ["npm run foo:fix", "npm run bar:fix", "npm run foo-bar:fix"]
+                    -> lintFixChildren = ["foo", "bar", "foo-bar"]
+            */
+            const lintRegex = /npm run (\w+[-:\w]*)/g;
+
+            // TODO: we should change the logic below to accept more flexible "lint" & "lint:fix" format
```

Comment:
Nit: It would probably be worth elaborating on what we mean by more flexible here. Specifically, we don't want to assume that the scripts are in the form "npm run foo && npm run bar".